src/main/kotlin/app/revanced/patcher/util/dom/DomUtil.kt

```diff
@@ -0,0 +1,15 @@
+package app.revanced.patcher.util.dom
+
+import org.w3c.dom.Node
+
+object DomUtil {
+    /**
+     * Recursively execute the [callback] on the [Node].
+     *
+     * @param callback The callback with the [Node].
+     */
+    fun Node.doRecursively(callback: (Node) -> Unit) {
```

Comment:
You *might* wanna add a limiter in here, to ensure a stack overflow doesn't happen (which can be hard to debug!).

Reply:
An iterative approach could be taken, though, what's the maximum depth of an XML file?

Reply:
AXML has a maximum depth of `2^16`, but the stack would overflow before we ever reach that.
I could not find anything in the XML spec that limits the depth.

Reply:
In that case, an iterative approach has to be taken, not a limiter.

Reply:
Actually, it appears the function was only used by split branch to merge resources. This means we can safely delete it